src/main/java/org/peergos/util/TraceLogger.java

```diff
@@ -253,25 +261,50 @@ private TraceLogger() {
         nodeIdMap.put("12D3KooWCfWmdJYdAUwm1pTxFKsRDRzGmsbeUMpriNNocMDVMmum", 9);
         currentNodeId = -1;
 
+        createAndSetupOutput();
+
+    }
+
+    private void createAndSetupOutput() {
         try {
-            outFile = new FileOutputStream(Client.DEFAULT_IPFS_DIR_PATH.toAbsolutePath().toString() + "/trace.log",
+            // Prepend each log file with the creation timestamp
+            long logFileCreationTimestampMillis = System.currentTimeMillis();
+
+            outFile = new FileOutputStream(
+                    Client.DEFAULT_IPFS_DIR_PATH.toAbsolutePath().toString() + "/" + logFileCreationTimestampMillis
+                            + ".trace.log",
                     /* append= */ true);
+
+            // An output stream with 100 MB buffer.
+            output = new BufferedOutputStream(outFile, 100 * 1024 * 1024);
+            flusher = new Thread(new Thread(new Runnable() {
+                public void run() {
+                    try {
+                        flushLogs();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }));
+            flusher.start();
         } catch (FileNotFoundException e) {
             e.printStackTrace();
             return;
         }
-        // An output stream with 100 MB buffer.
-        output = new BufferedOutputStream(outFile, 100 * 1024 * 1024);
-        flusher = new Thread(new Thread(new Runnable() {
-            public void run() {
-                try {
-                    flushLogs();
-                } catch (IOException e) {
-                    e.printStackTrace();
+    }
+
+    private void _writeLog(String log) {
```

Comment:
Consider moving this method below writeLog()

Reply:
Good idea, appreciate these style pieces